include/LeMonADE/feature/FeatureConnectionSc.h

```diff
@@ -0,0 +1,621 @@
+////////////////////////// member definitions /////////////////////////////////
+/**
+ * @details The function is called by the Ingredients class when an object of type Ingredients
+ * is associated with an object of type FileImport. The export of the Reads is thus
+ * taken care automatically when it becomes necessary.\n
+ * Registered Read-In Commands:
+ * * !reactivity
+ *
+ * @param fileReader File importer for the bfm-file
+ * @param destination List of Feature to write-in from the read values.
+ * @tparam IngredientsType Features used in the system. See Ingredients.
+ **/
+template<class IngredientsType>
+void FeatureConnectionSc::exportRead(FileImport< IngredientsType >& fileReader)
+{
+  fileReader.registerRead("!reactivity",new ReadReactivity<IngredientsType>(fileReader.getDestination()));
+}
+
+
+/**
+ * The function is called by the Ingredients class when an object of type Ingredients
+ * is associated with an object of type AnalyzerWriteBfmFile. The export of the Writes is thus
+ * taken care automatically when it becomes necessary.\n
+ * Registered Write-Out Commands:
+ * * !reactivity
+ *
+ * @param fileWriter File writer for the bfm-file.
+ */
+template<class IngredientsType>
+void FeatureConnectionSc::exportWrite(AnalyzerWriteBfmFile< IngredientsType >& fileWriter) const
+{
+  fileWriter.registerWrite("!reactivity",new WriteReactivity<IngredientsType>(fileWriter.getIngredients_()));
+}
+
+/******************************************************************************/
+/**
+ * @fn bool FeatureConnectionSc::checkMove( const IngredientsType& ingredients, const MoveConnectSc& move )const
+ * @brief Returns true for all moves other than the ones that have specialized versions of this function.
+ * This dummy function is implemented for generality.
+ * @details  it might make a difference for the speed if the order of statements is switched for different systems parameters
+ * @param [in] ingredients A reference to the IngredientsType - mainly the system
+ * @param [in] move General move other than MoveLocalSc or MoveLocalBcc.
+ * @return true Always!
+ */
+/******************************************************************************/
+template<class IngredientsType>
+bool FeatureConnectionSc ::checkMove(const IngredientsType& ingredients, const MoveConnectSc& move) const
+{
+	uint32_t ID(move.getIndex());
+	VectorInt3 dir(move.getDir());
+	VectorInt3 Pos(ingredients.getMolecules()[ID]);
+	//check if there is a monomer to connect with
+	if ( connectionLattice.getLatticeEntry(Pos+dir) == 0 ) return false;
+	//check for maximum number of bonds for the first monomer
+	if ( ingredients.getMolecules()[ID].getNMaxBonds() >= ingredients.getMolecules().getNumLinks(ID) ) return false;
+	uint32_t Neighbor(connectionLattice.getLatticeEntry(Pos+dir)-1);
+	//check for maximum number of bonds for the second monomer
+	if ( ingredients.getMolecules()[Neighbor].getNMaxBonds() >= ingredients.getMolecules().getNumLinks(Neighbor) ) return false;
```

Comment:
**STRONG LOGICAL ERROR**
should read 
`if ( ingredients.getMolecules().getNumLinks(Neighbor) >= ingredients.getMolecules()[Neighbor].getNMaxBonds()  )`